apps/application-system/api/src/app/modules/application/application.controller.ts

```diff
@@ -237,7 +239,24 @@ export class ApplicationController {
       typeId: application.typeId,
     }
 
-    return this.applicationService.create(applicationDto)
+    const createdApplication = await this.applicationService.create(
+      applicationDto,
+    )
+
+    // Make sure the application has the correct lifecycle values persisted to database.
+    // Requires an application object that is created in the previous step.
+    const {
+      updatedApplication,
+    } = await this.applicationService.updateApplicationState(
+      createdApplication.id,
```

Comment:
Not related to your code, but I wonder if passing the Application object would be easier than each needed property as parameter